fedimint-core/src/api.rs

```diff
@@ -396,6 +406,43 @@ pub trait FederationApiExt: IRawFederationApi {
         )
         .await
     }
+
+    async fn request_admin<Ret>(
+        &self,
+        method: &str,
+        params: ApiRequestErased,
+        auth: ApiAuth,
+    ) -> FederationResult<Ret>
+    where
+        Ret: serde::de::DeserializeOwned + Eq + Debug + Clone + MaybeSend,
+    {
+        // We would never want to accidentally send our password to everyone
+        assert_eq!(
+            self.all_peers().len(),
+            1,
+            "attempted to broadcast admin password?!"
+        );
+        self.request_current_consensus(method.into(), params.with_auth(auth))
```

Comment:
Could we use `request_single_peer` here instead of `request_current_consensus`?

Reply:
It makes sense, but it changes the type of the result, which will lead further changes, and I have other PRs on top of this one, and just wanted to move these methods 1:1 to a new place to keep it small. We'll follow up with improvements. Added a TODO to #4519 4519

Reply:
Works for me